### PR TITLE
refactor(Core/acore_string): Split locale

### PR DESCRIPTION
**Pull request by pangolp**
## Changes Proposed:
-  I propose to divide the table in 2. The first one will contain the English texts, and the second one will be composed of all the translations that the community will contribute.

- Currently, it is very difficult to separate and update the information in the table. If we manage to standardize it, we could "eliminate" those languages that we may not need. As well, you can update, insert or delete records in a simpler way. My intention is not to eliminate languages, on the contrary, is to add more and more, but maybe sometimes, some tables have many records and I thought it was a good idea to divide the content.

## Issues Addressed:

- Closes

## Discussions
- https://github.com/azerothcore/azerothcore-wotlk/discussions/6411

## Tests Performed:

- Windows 10

## How to Test the Changes:

At the moment, there is nothing to test, because I have only built from SQL.

## Screenshots

acore_string

acore_string_locale

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by pangolp** [comment]
@buddiman In case you manage to finish this Pull request and people like the idea, your command should be updated. I think it is an interesting idea, mainly to free some space for languages that we don't need, as well as to add new translations. In the case of Spanish, there is only 1 translation in the whole table. I would love to know your opinion on this, given that you were working on the command and that it seems to be coming to the emulator any day now.

File: data/sql/updates/pending_db_world/rev_1623976434519989200.sql
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1623976434519989200.sql:8]: Please read https://www.azerothcore.org/wiki/sql-standards

**Review by Kitzunu — CHANGES_REQUESTED** [review]

File: data/sql/updates/pending_db_world/rev_1623976434519989200.sql
Comment (pangolp) [reply data/sql/updates/pending_db_world/rev_1623976434519989200.sql:8]: > Please read https://www.azerothcore.org/wiki/sql-standards

Ok, I will read it, but it would be good, if you could quote me an example please, because in the documentation there is none about creating a table. The only thing I did, was to take the examples found in other PR.

Comment (Kitzunu) [reply data/sql/updates/pending_db_world/rev_1623976434519989200.sql]: There is a Header about Tables and Columns... https://www.azerothcore.org/wiki/sql-standards#tables-and-columns

Comment (pangolp) [reply data/sql/updates/pending_db_world/rev_1623976434519989200.sql]: I already made a commit with the fix, what is the error?

Comment (Kitzunu) [reply data/sql/updates/pending_db_world/rev_1623976434519989200.sql]: It is good now.

Comment (pangolp) [reply data/sql/updates/pending_db_world/rev_1623976434519989200.sql]: Thank you.

**Comment by pangolp** [comment]
I am having trouble organizing the code. I managed to return the text in English, that is, the original, but I can't get it to take the information from `acore_string_locale`, but I'll see what I can come up with.

**Comment by Winfidonarleyan** [comment]
Need change loading process 

> I am having trouble organizing the code. I managed to return the text in English, that is, the original, but I can't get it to take the information from `acore_string_locale`, but I'll see what I can come up with.

**Comment by pangolp** [comment]
> Need change loading process
> 
> > I am having trouble organizing the code. I managed to return the text in English, that is, the original, but I can't get it to take the information from `acore_string_locale`, but I'll see what I can come up with.

I'm a bit confused really with the situation, I thought that by reviewing the old pull request, where the updates were made in the other tables I could do it, but the truth is that I'm a bit confused. It seems like an interesting thing to me, because we could remove maybe those languages that we don't need, and also, it's a little easier to maintain. But I am not being able to do the loading of the translations.

I was able to modify the original table and the method that loads the information, but I can't create the other method, the one that loads the translations, I'm a bit confused. If you want I can make a commit, with the modification of the existing method and tell me if it is correct, that would be a great help.

**Comment by pangolp** [comment]
Sorry I uploaded it that way, but I wanted you to see the logic I've more or less applied so far. Actually, it is very likely that the code in this way does not compile. But I wanted to upload it anyway, so that someone with a little more experience can tell me where I'm going wrong and correct it. Thanks.

**Comment by pangolp** [comment]
I think it works, but it would have to be tested with other non-Spanish clients.

WoWScrnShot_062421_195212

WoWScrnShot_062421_195340

File: src/server/game/Globals/ObjectMgr.cpp
Comment (pangolp) [reply src/server/game/Globals/ObjectMgr.cpp:7907]: Now I'll fix it. Yesterday I had a conflict, and I wasn't sure which option to use. I must have read it wrong, so I used the current one.

**Comment by pangolp** [comment]
If the Pull request is not approved, I will understand, because I know that perhaps the motivation for it may not be well understood. But in reality, it helped me more than anything else, to understand a little bit some things that I didn't understand well how it worked. Or maybe later on, it seems like a good idea to do it, and we already have the code. In that aspect, if you don't agree with the changes, I will understand.

**Comment by FrancescoBorzi** [comment]
@azerothcore/testers @azerothcore/developers can anyone please check this?

**Comment by pangolp** [comment]
Before doing this merge, if you want to do it, we should review the other PR, where I uploaded about 300 translations, because if not, it will be obsolete and will not be useful. Anyway, this PR is going to take a while to test maybe. The other one I think is ready to be merged. https://github.com/azerothcore/azerothcore-wotlk/pull/6529

**Comment by FrancescoBorzi** [comment]
what's the status of this PR?

**Comment by pangolp** [comment]
>  what's the status of this PR?

I don't know. I worked on it a long time ago. The idea was to split the board into 2. But that would make a lot of addons and things no longer work. However, we could easily delete, add or update new texts. But I don't know the current status of this application.

**Comment by FrancescoBorzi** [comment]
@pangolp ok then closing this for now. If you want to continue working on it, we can always reopen it.

**Comment by pangolp** [comment]
> ok then closing this for now. If you want to continue working on it, we can always reopen it.

As far as I remember, there was nothing to add. The only thing left to do was to test in other languages and with other clients than Spanish and decide whether or not they wanted the tables to be divided in 2. Because basically it is something that I came up with, but if it is not approved or if it is not uploaded, it is really indistinct, because anyway it helped me to understand some things. It just seemed to me a good idea, to split the tables, so that those projects, that don't use so many game clients, can eliminate the languages that they don't want to have.

**Comment by pangolp** [comment]
In other words, it is a suggestion, but maybe not everyone will share it, I just found interesting the fact of being able to divide the tables in 2. The one with all the texts in English, and the one with the translations, like many other tables in the game. The problem is that many addons may stop working, because if they are not updated, they will look for a table that does not exist.

**Comment by pangolp** [comment]
I think it would be a good idea to reopen this topic, because the truth is that the standardization of the table was applied to many others in the game. However, I think this is the only one that has not been normalized yet. What I'm not sure is how to reopen this, since I deleted the table, and I don't know if it would be better to reopen it in another pull request, because it's been a long time and I don't know how many conflicts I have to solve.

**Pull request closed without merging**
